lib/auth/google_oauth2_authenticator.rb

```diff
@@ -32,6 +32,9 @@ def register_middleware(omniauth)
           opts[:client_id] = SiteSetting.google_oauth2_client_id
           opts[:client_secret] = SiteSetting.google_oauth2_client_secret
 
+          # NOTE: uncomment when testing locally
+          # opts[:redirect_uri] = "http://localhost:4200/auth/google_oauth2/callback"
```

Comment:
Took me a while to figure that one out :|

Reply:
I think we can fix this properly: https://github.com/discourse/discourse/pull/33102

(and then drop these comments)